feat(ISV-5571): ignore SBOM-related task errors in rh-advisories

wcheang opened the pull request:
## Describe your changes
Failing to process or upload SBOMs should not fail the whole rh-advisories pipeline. This change ensures that the pipeline will continue even if there are errors in the SBOM-related tasks.

## Relevant Jira
https://issues.redhat.com/browse/ISV-5571

## Checklist before requesting a review
- [x] I have marked as draft or added `do not merge` label if there's a dependency PR 
  - If you want reviews on your draft PR, you can add reviewers or add the `release-service-maintainers` handle if you are unsure who to tag
- [x] My commit message includes `Signed-off-by: My name <email>`
- [x]  I have bumped the task/pipeline version string and updated changelog in the relevant README
- [x] I read CONTRIBUTING.MD and [commit formatting](CONTRIBUTING.md#commit-message-formatting-and-standards)

mmalina commented on pipelines/managed/rh-advisories/rh-advisories.yaml:
I don't think it's right to ignore issues with sbom upload - I think this is a mandatory part of the release. I commented on the Jira to clarify this.

johnbieren replied:
If the decision does become to ignore issues with sbom upload (not advocating for this, just saying if), then I think the proper way to accomplish this would be to modify the tasks to output a clear error message and exit 0 instead of failing for these scenarios. I am worried about the user experience if you have a failed pipelinerun because it failed in... embargo-check for example, but there are also now these sbom tasks that show as failing. I bet many users will be confused and thinking the pipeline failed from the sbom tasks

mmalina replied:
There was a reply in Jira and it seems this is a conscious decision. Atlas is not really used in production yet and also, they plan to implement a mechanism to add sbom data post release in case it doesn't succeed during release. So I'm fine with the intention of this change.

But I do agree with Johnny's suggestion. ~One option would be to display a warning for the task in Konflux UI, similar to what EC does. But I am not sure how that's done (because AFAIK, Tekton doesn't have anything like that). So I asked the EC team here: https://redhat-internal.slack.com/archives/C02CTEB3MMF/p1738052702684559~

Edit: The "warning" UI feature is really not suited for this use case - it is designed to be used by tasks that run some sort of tests and then you would provide test results via a result called TEST_OUTPUT and then one of the possible outcomes of a test is a warning.

wcheang replied:
Thanks for the feedback and good points about user experience. I updated to limit the change to only the actual curl command that sends requests to Atlas, and output an message without indicating any failures, since the main point of this change is to accommodate any Atlas instability. If other SBOM-related tasks fail, it might indicate that there's something wrong with the data, in which case we'd probably still want the pipeline to fail.

wcheang committed:
feat(ISV-5571): ignore SBOM-related task errors in rh-advisories

Instability in Atlas should not fail the whole pipeline. This change
ensures that the pipeline will continue even if the request to upload
to Atlas fails.

Signed-off-by: Wai Cheang <wcheang@redhat.com>

johnbieren reviewed (dismissed):
I am cool with this change, but let's see if Martin is too. It would be nice to set this to yellow (I think that is the unstable status in tekton?) in the pipeline visualization to alert users something happened here, but given I don't know how to do that, not too fair for me to ask you to :slightly_smiling_face:

johnbieren commented:
/ok-to-test

jedinym commented:
@johnbieren any idea why the tests still haven't started?

johnbieren commented:
Hmm, no, but retest should start them

johnbieren commented:
/retest

mmalina commented:
> I am cool with this change, but let's see if Martin is too. It would be nice to set this to yellow (I think that is the unstable status in tekton?) in the pipeline visualization to alert users something happened here, but given I don't know how to do that, not too fair for me to ask you to 🙂

I don't think Tekton has any option to show a taskrun as yellow. AFAIK, it's a Konflux feature, but not really applicable for our use case, see my comment here: https://github.com/konflux-ci/release-service-catalog/pull/786#discussion_r1931717628

mmalina commented on tasks/managed/upload-sbom-to-atlas/upload-sbom-to-atlas.yaml:
```diff
@@ -229,6 +229,7 @@ spec:
             --data "@$supported_version_of_sbom" \
-            "$bombasticApiUrl/api/v1/sbom?id=$sbom_id"
+            "$bombasticApiUrl/api/v1/sbom?id=$sbom_id" \
+            || (echo "SBOM upload to Atlas has failed!")
```
Why the `()`? That opens a subshell, is that necessary? It seems to work just fine without it.

Also, would be nice to have a test for this - to mock curl to return a non-zero code and confirm the task still passes.

wcheang replied:
Removed the `()` and updated the `mocks.sh` to return the error code for curls to Atlas.

mmalina replied:
Thanks. But it would be preferable if this was a separate test - the way to do it is create a new test pipeline and in it use a special string for the curl call - in this case the easiest would probably be to use a file like `"$sbomsDir/spdx_minimal_fail_2_2.json"` and look for `spdx_minimal_fail` in the mock or something like that.

wcheang replied:
No problem. Added new test.

mmalina replied:
Thanks.

mmalina commented:
/ok-to-test

mmalina commented:
/retest

mmalina commented:
BTW, my e2e run just failed on atlas error :)
```
PipelineRun managed-rvrh5 failed
Expected
    : Pipelinerun 'managed-rvrh5' didn't succeed\nLogs from failed container 'managed-rvrh5-upload-product-sbom/step-upload-sboms': \n--- Processing test-product-v1.0.json ---\nGetting SSO token from https://auth.stage.redhat.com/auth/realms/EmployeeIDP/protocol/openid-connect/token\nUploading SBOM to https://sbom.atlas.release.stage.devshift.net (with id=test-product-v1.0)\ncurl: (22) The requested URL returned error: 500\nInternal server error
to equal
    : 
```
Here: https://github.com/konflux-ci/release-service-catalog/pull/807

jedinym commented:
@wcheang Do you have a clue why the e2e run above failed? It seems weird.

wcheang commented:
@jedinym no idea, we don't have any alerting enabled for stage Atlas and the migration last week was on Feb 7 not Feb 6. But since it's stage I don't think stability/availability are currently guaranteed anyway. It shouldn't be failing e2e tests anymore though, since this ignore error change is merged.

jedinym commented:
@wcheang Oh, I thought that it failed after the merge. Nevermind then.
